### DISCUSSION TITLE
Does `nvim_del_autocmd` work?

### DISCUSSION THREAD

**Discussion by mangelozzi** [neovim/neovim#29787] [General]
If you run neovim with this as your init.lua:

```lua
local MyGroup = vim.api.nvim_create_augroup("MyGroup", {clear = true})
local Done = false
vim.api.nvim_create_autocmd(
    "BufEnter",
    {
        group = MyGroup,
        callback = function()
            print("ENTER BUF!")
            if not Done then
                Done = true
                vim.api.nvim_del_autocmd(MyGroup)
                print('Deleted autocmd')
            end
        end
    }
)
```

It will continually print "ENTER BUF!" when ever you enter a new buffer, even after printing `Deleted autocmd`. Maybe I don't understand what `nvim_del_autocmd`, the help just says `Deletes an autocommand by id.` ... Which I would except to stop the autocmd from executing?

Is this a bug? Im using the nightly build.

**Comment by zeertzjq** [comment]
You should use `nvim_del_augroup_by_id`

**Reply by mangelozzi** [reply]
Oh my, thank you so much, I thought it had to be me and not bug. I see the difference now between `augroup` and `aucmd`, thanks so much!
